fingerprint.c

```diff
@@ -207,18 +213,23 @@ void *auth_thread_loop(void *arg)
                     msg.data.authenticated.hat = hat;
 
                     callback(&msg);
+
+                    fpc_auth_end(sdev->fpc);
+                    pthread_mutex_lock(&sdev->lock);
+                    sdev->worker.thread_running  = false;
+                    pthread_mutex_unlock(&sdev->lock);
                     break;
                 }
             }
+            fingerprint_msg_t msg;
```

Comment:
why send the message twice and once with bad/weird data? also this sends authenticated message even when it's not authenticated?

Reply:
I saw it in biometrics service, when you send authenticated message with fid=0 it means not authenticated.